### PR TITLE
feat: add update functions

### PR DESCRIPTION
**Pull request by bdraco**
Tested systems
- [x] R3.9.36 (a1a93f) 192.168.106.159 (89 nem)
- [x] D3.17.3 (0e8c7a) 192.168.121.200 (81 nem)
- [x] D5.0.62 (5e57a9) 192.168.111.40 (12 nem)
- [ ] D7.3.130 (91f206) 192.168.107.127 (89 nem plus) -[ not part of this PR](https://github.com/pyenphase/pyenphase/pull/17)

### CODE REVIEW

File: src/pyenphase/envoy.py [hunk 1]
```diff
@@ -45,8 +49,29 @@ def create_no_verify_ssl_context() -> ssl.SSLContext:
...
+    MODEL_S = "PC"
+    MODEL_C = "P"
+    MODEL_LEGACY = "P0"
+
+
+ENDPOINT_URL_PRODUCTION_V1 = "/api/v1/production"
+ENDPOINT_URL_PRODUCTION_JSON = "/production.json"
+ENDPOINT_URL_PRODUCTION = "/production"
+
+ENDPOINT_URL_PRODUCTION_INVERTERS = "/api/v1/production/inverters"
```
Comment (cgarwood) [inline src/pyenphase/envoy.py:70]: Most of these are in `const` already

Comment (bdraco) [reply src/pyenphase/envoy.py]: adjusted 👍

**Comment by bdraco** [comment]
I can't test with the 7.x system because the wifi signal is too bad. I'll have to run ethernet to it but I'm out of outdoor ethernet so I'll have to run into town to get some and than pull a new line later today.

In the mean time, 3.x/5.x is all working

**Review by joostlek — APPROVED** [review]

**Review by dgomes — APPROVED** [review]

**Comment by bdraco** [comment]
Will fix the ci when I get back home

**Pull request merged**